### DISCUSSION TITLE
Where do I get the HAID from?

### DISCUSSION THREAD

**Discussion by GRX** [alexkn/node-red-contrib-homeconnect#40] [Q&A]
The docs guide me to get Node-red authenticated, and then asks for a HAID.
Nowhere in the docs does it state what the HAID is, or where to retrieve it from.

I'm not a HomeConnect developer, I just want to handle events when my appliances are finished.

**Comment by GRX** [comment]
I thought it should be done via the Request node. but no matter what I do I just get back: `auth not ready`

**Comment by Richardmarobert** [comment]
You may want to look into the API, which is on the Home Connect developers site. You have the option to register the application, but you could also select the API, and see all the appliances registered on your HomeConnect account. The HAID is to be found there, as well.
